js/src/wrappers/vercel/exporter.ts

```diff
@@ -0,0 +1,647 @@
+    const data: LangChainMessageFields = { content: message.content };
+
+    if (Array.isArray(message.content)) {
+      data.content = message.content.map((part) => {
+        if (part.type === "text") {
+          return {
+            type: "text",
+            text: part.text,
+            ...part.experimental_providerMetadata,
+          };
+        }
+
+        if (part.type === "tool-call") {
+          return {
+            type: "tool_use",
+            name: part.toolName,
+            id: part.toolCallId,
+            input: part.args,
+            ...part.experimental_providerMetadata,
+          };
+        }
+
+        return part;
+      });
+
+      const toolCalls = message.content.filter(
+        (part): part is ToolCallPart => part.type === "tool-call"
+      );
+
+      if (toolCalls.length > 0) {
+        data.additional_kwargs ??= {};
+        data.additional_kwargs.tool_calls = toolCalls.map((part) => {
+          return {
+            id: part.toolCallId,
+            type: "function",
+            function: {
+              name: part.toolName,
+              id: part.toolCallId,
+              arguments: JSON.stringify(part.args),
+            },
+          };
+        });
+      }
+    }
+
+    return { type: "ai", data };
+  }
+
+  if (message.role === "user") {
+    const data: LangChainMessageFields = { content: message.content };
+
+    if (Array.isArray(message.content)) {
+      data.content = message.content.map((part) => {
+        if (part.type === "text") {
+          return {
+            type: "text",
+            text: part.text,
+            ...part.experimental_providerMetadata,
+          };
+        }
+
+        if (part.type === "image") {
+          return {
+            type: "image_url",
+            image_url: part.image,
+            ...part.experimental_providerMetadata,
+          };
+        }
+
+        return part;
+      });
+    }
+
+    return { type: "human", data };
+  }
+
+  if (message.role === "system") {
+    return { type: "system", data: { content: message.content } };
+  }
+
+  if (message.role === "tool") {
+    const res = message.content.map((toolCall) => {
+      return {
+        type: "tool",
+        data: {
+          content: JSON.stringify(toolCall.result),
+          name: toolCall.toolName,
+          tool_call_id: toolCall.toolCallId,
+        },
+      };
+    });
+    if (res.length === 1) return res[0];
+    return res;
+  }
+
+  return message;
+}
+
+const tryJson = (
+  str:
+    | string
+    | number
+    | boolean
+    | Array<null | undefined | string>
+    | Array<null | undefined | number>
+    | Array<null | undefined | boolean>
+    | undefined
+) => {
+  try {
+    if (!str) return str;
+    if (typeof str !== "string") return str;
+    return JSON.parse(str);
+  } catch {
+    return str;
+  }
+};
+
+function stripNonAlphanumeric(input: string) {
+  return input.replace(/[-:.]/g, "");
+}
+
+function convertToDottedOrderFormat(
```

Comment:
AFAICT, the order in which the spans arrive to the trace exporter matches the "execution order",